Python.md

```diff
@@ -40,7 +85,37 @@ result = connection.execute(sqlalchemy.text("""
            show_gravatar
       FROM "user"
      WHERE musicbrainz_id IN :musicbrainz_usernames
-"""), {
-    'musicbrainz_usernames': tuple(usernames),
-})
+""")
+result = connection.execute(query, {'musicbrainz_usernames': tuple(usernames)})
 ```
+
+It is preferred to assign the query to a separate variable and not include it inside the
+`execute` call.
+
+TODO: Projects which don't use sqlalchemy
+
+### Tests
+
+You should write tests for all new methods that you write. If you update a method, make sure that
+tests for that method still pass, and write/remove tests as necessary for the change in functionality.
+
+Tests for a module (`package/module.py`) go in a `test` subdirectory - `package/test/test_module.py`
+
+Database tests (for modules in `db`) can use the database. Inherit from `db.testing.DatabaseTestCase`
+to automatically reset the database at the end of each testcase.
```

Comment:
I guess I'm not sure what you are trying to explain here. Let's say there are no mentions of modules/packages:

> Database tests can use the database. Inherit from a base testing package that automatically resets the database at the end of each testcase.

The only part I can see being useful is about putting common functionality from tests into a base test case class, but this is not specific to databases or project.

What I'm suggesting is that it might not be a good idea to make assumptions about how to structure specific part of a project. This seems unnecessarily restrictive. I would, however, agree with adding more general suggestions about testing and coding.